calblueprint/girls-write-now
# [auth] Styled and implemented forgotPassword screen

[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Styled forgotPassword screen to include a header, input for email or username, subtext, error messages if username/email isn't found, and button that routes to signUp page
- Error messages display until error is solved
- Button is disabled until email/username is valid 

## Relevant Links

### Notion Sprint Task

[//]: # 'Add the relevant Notion link(s) here'

[Sprint Page](https://www.notion.so/calblueprint/auth-Style-forgot-password-request-b20f87fd9fa14341951880b13b91f1de?pvs=4)

### Online sources

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

[Mutex documentation](https://www.npmjs.com/package/async-mutex)

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

[//]: # 'The order in which to review files and what to expect when testing locally'

- Create an account/use an existing one
- Go to forgot password screen, attempt entering account's username/email
- Check that error message displays and "Continue" button is disabled until email/username is valid
- Check that "Continue" button routes you to the sign up page

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # 'Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally'

- Created new account and went through forgot password flow, managed to successfully reset password
- Tried entering invalid emails and usernames, error messages are working 

### Screenshots

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

![IMG_2470](https://github.com/calblueprint/girls-write-now/assets/156966341/77881f07-872b-402f-92d0-91103570f868)
![IMG_2471](https://github.com/calblueprint/girls-write-now/assets/156966341/ddefb942-d028-4534-86cf-580e138261fc)
![IMG_2472](https://github.com/calblueprint/girls-write-now/assets/156966341/fa5c9a9d-1650-471f-89c1-a91044e7c080)
![IMG_2473](https://github.com/calblueprint/girls-write-now/assets/156966341/3d4b7482-8a5e-4138-974c-c0a0e9f24d1d)

CC: @adityapawar1

[//]: # 'This tags in Adi as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
